### PR TITLE
[Docs] DG messages

### PR DESCRIPTION
**Pull request by kelvinou01**

### CODE REVIEW

**Review by Polygonalr — COMMENTED** [review]
LGTM, not needed but probably can use sequence diagram to explain Step 1 also.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -246,12 +246,64 @@ _{more aspects and alternatives to be added}_
...
+
+Step 3: The user realises his first attempt at a pitch isn't working well, so they delete the message from the address book. `DeleteMessageCommand#execute()` calls `ModelManager#deleteMessage()`, which calls `AddressBook#deleteMessage()`, which deletes the message from `AddressBook#messages`
+
+```
+message delete 1
+```
 
 #### Design considerations:
 
+- **Aspect: Allow edting**
```
Comment (malwaregarry) [inline docs/DeveloperGuide.md:293]: You might have missed a typo here
```suggestion
- **Aspect: Allow editing**
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -246,12 +246,64 @@ _{more aspects and alternatives to be added}_
...
+
+The following commands are provided:
+
+`create`  — Create a message template
+
+`delete`  — Delete the specified message template
+
+`generate`  — Using the specified message template and client, generate a message for client. 
+
+Creation and deletion are exposed in the `Model` interface as `Model#addMessage`, `Model#deleteMessage`, while message generation is exposed in `Message` as `Message#generate`.
```
Comment (weiquany) [inline docs/DeveloperGuide.md:263]: I think it is `Model#createMessage`.

**Review by craeyeons — COMMENTED** [review]
LGTM!

**Pull request merged**